src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```
@@ -128,13 +128,27 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
             services.AddSingleton<IFunctionMetadataManager, FunctionMetadataManager>();
             services.AddSingleton<IFunctionMetadataProvider, HostFunctionMetadataProvider>();
             services.AddSingleton<IWebFunctionsManager, WebFunctionsManager>();
-            services.AddSingleton<IInstanceManager, InstanceManager>();
             services.AddHttpClient();
             services.AddSingleton<StartupContextProvider>();
             services.AddSingleton<IFileSystem>(_ => FileUtility.Instance);
             services.AddTransient<VirtualFileSystem>();
             services.AddTransient<VirtualFileSystemMiddleware>();
 
+            if (SystemEnvironment.Instance.IsLinuxConsumptionOnAtlas())
+            {
+                System.Console.WriteLine("[TEST] Atlas Linux Consumption");
+                services.AddSingleton<LinuxInstanceManager, InstanceManager>();
+            }
+            else if (SystemEnvironment.Instance.IsLinuxConsumptionOnLegion())
+            {
+                System.Console.WriteLine("[TEST] Legion Linux Consumption");
+                services.AddSingleton<LinuxInstanceManager, LegionInstanceManager>();
+            }
+            else
+            {
+                System.Console.WriteLine("[TEST] Else");
```

Should we throw here? Or can we have a default, like assume Atlas if unable to tell from environment? This will retain previous behavior.

Drawback of that is it will hide any issues where we expect the environment to tell us legion or atlas, but it isn't, and we just continue on assuming it is Atlas (when that may be incorrect).

This boils down to: do we have confidence all environments will correctly report legion vs atlas (and so we follow fail fast principal and throw when it is not clear), or will we require a fallback behavior here (to Atlas) when environment details are not available?

---

Made it default to AtlasInstanceManager since that was the previous InstanceManager injected. The IsLinuxConsumptionOnAtlas and IsLinuxConsumptionOnLegion checks should be accurate, but if they cause an issue, we should change the checks and leave the dependency injection logic